apps/condo/domains/ticket/components/TicketForm/UpdateTicketForm.tsx

```diff
@@ -100,22 +103,73 @@ export const UpdateTicketForm: React.FC<IUpdateTicketForm> = ({ id }) => {
     const intl = useIntl()
 
     const { replace } = useRouter()
+    const { persistor } = useCachePersistor()
     const { obj, loading: ticketLoading, refetch, error } = Ticket.useObject({ where: { id } })
     const { objs: files, refetch: refetchFiles } = TicketFile.useObjects({ where: { ticket: { id } } })
     const { objs: invoices, loading: invoicesLoading } = Invoice.useObjects({ where: { ticket: { id } } })
+    const {
+        data: observersData,
+        loading: observersLoading,
+    } = useGetTicketObserversByTicketIdQuery({
+        variables: {
+            ticketId: id,
+        },
+        skip: !id || !persistor,
+    })
+    const observers = useMemo(() => observersData?.observers?.filter(Boolean) || [], [observersData?.observers])
+
+    const ticketOrganizationId = useMemo(() => get(obj, ['organization', 'id']) || null, [obj])
+
+    const { 
+        data: employeesData,
+        loading: employeesLoading,
+    } = useGetEmployeesByOrganizationIdAndUserIdsQuery({
+        variables: {
+            organizationId: ticketOrganizationId,
+            userIds: observers.map(o => o?.user?.id).filter(Boolean),
+        },
+        skip: !ticketOrganizationId || !persistor || observers.length === 0,
+    })
+
+    const allowedObserverUserIds = useMemo(() => {
+        const employees = employeesData?.employees?.filter(Boolean) || []
+        return new Set(employees.map(e => e?.user?.id).filter(Boolean))
+    }, [employeesData])
 
     // no redirect after mutation as we need to wait for ticket files to save
     const [action] = useUpdateTicketMutation({})
     const createInvoiceAction = Invoice.useCreate({})
     const updateInvoiceAction = Invoice.useUpdate({})
+
     const updateAction = async (values) => {
-        const { existedInvoices, newInvoices, ...ticketValues } = values
+        const { existedInvoices, newInvoices, observers: newObserverUserIds, ...ticketValues } = values
+
+        const newObserverUserIdsArray = Array.isArray(newObserverUserIds) ? newObserverUserIds.filter((userId) => typeof userId === 'string') : []
+
+        const existingAllowedUserIdSet = new Set(observers.map((o) => o?.user?.id).filter(Boolean))
```

Comment:
We have a similar-ish logic in Scopes creation in News domain. Perhaps we can create a shared function?